### PR TITLE
Resource Group resource and datasource

### PR DESCRIPTION
**Pull request by Krunal-Thakkar**
# Description
Resource Group resource and datasource

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: powerflex/provider/service_resource_schema.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ import (
 
 // ServiceReourceSchema - variable holds schema for Service resource
 var ServiceReourceSchema schema.Schema = schema.Schema{
+	DeprecationMessage:  "Use Resource Group Datasource instead",
```
Comment (shenda1) [inline powerflex/provider/service_resource_schema.go:34]: Resource

File: docs/resources/resource_group.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+---
+# Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (shenda1) [inline docs/resources/resource_group.md:2]: Please check if copyright year will be 2024 or 2023-2024?

File: examples/data-sources/powerflex_resource_group/data-source.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# commands to run this tf file : terraform init && terraform apply --auto-approve
+
+# Get all Resource Group details present on the cluster
+data "powerflex_resource_group" "example1" {
+}
+
+# Get Resource Group details using the ID of the service.
+data "powerflex_resource_group" "example2" {
+  service_ids = ["ID1", "ID2"]
```
Comment (AnikaAgiwal2711) [inline examples/data-sources/powerflex_resource_group/data-source.tf:26]: This will remain as service_ids or resource_group_ids ??

**Review by shenda1 — APPROVED** [review]

File: examples/data-sources/powerflex_resource_group/data-source.tf (hunk 3 above)
Comment (Krunal-Thakkar) [reply examples/data-sources/powerflex_resource_group/data-source.tf:26]: need to change according to resource name

**Comment by Krunal-Thakkar** [comment]
run e2e test

File: examples/data-sources/powerflex_resource_group/data-source.tf (hunk 3 above)
Comment (Krunal-Thakkar) [reply examples/data-sources/powerflex_resource_group/data-source.tf:26]: @gokul-srivathsan could you please confirm?

Comment (gokul-srivathsan) [reply examples/data-sources/powerflex_resource_group/data-source.tf:26]: @Krunal-Thakkar  we need to change to resource_group_ids and resource_group_names.

File: docs/data-sources/resource_group.md [hunk 4]
```diff
@@ -48,21 +48,21 @@ limitations under the License.
 
 # commands to run this tf file : terraform init && terraform apply --auto-approve
 
-# Get all Resource Group details present on the cluster
+# Get all Resource Group details present on the Resource Group
```
Comment (AnikaAgiwal2711) [inline docs/data-sources/resource_group.md:51]: This should be cluster. Right ?

**Comment by Krunal-Thakkar** [comment]
E2e Success Run : https://osj-sio-03-prd.cec.lab.emc.com:8080/job/Terraform-Pipelines/job/PowerFlex/job/tf-powerflex-pr/187/

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Pull request merged**